### PR TITLE
Add a simulator for the RAM Hopper

### PR DESCRIPTION
**Pull request by rsinnet**
This allows callers of the hopper service to think the hopper is there.

### CODE REVIEW

File: miso_peripherals/nodes/ram_hopper_simulator_node [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+)
+import rospy
+
+
+class RAMHopperSimulatorNode(object):
+    """Simulates a RAM Hopper controller."""
+
+    def __init__(self):
+        """Create a new RAMHopperSimulatorNode."""
+        rospy.Service('/hopper/dump', HopperDump, self.__hopperdump)
```
Comment (zzv2) [inline miso_peripherals/nodes/ram_hopper_simulator_node:17]: this should be a constant shared by the real and sim nodes

Comment (rsinnet) [reply miso_peripherals/nodes/ram_hopper_simulator_node:17]: tru tru

Comment (rsinnet) [reply miso_peripherals/nodes/ram_hopper_simulator_node]: How about now? I refactored the whole thing.

Comment (zzv2) [reply miso_peripherals/nodes/ram_hopper_simulator_node]: it should probably still be a constant lol but I guess its fine

Comment (rsinnet) [reply miso_peripherals/nodes/ram_hopper_simulator_node]: Fixed.

File: miso_peripherals/src/miso_peripherals/hopper_host.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    def __init__(self, controller):
+        """Create a new HopperControllerNode."""
+        self.__controller = controller
+        rospy.Service(self.DUMP_SRV, HopperDump, self.__hopperdump)
+
+    def __hopperdump(self, req):
+        """Return successful."""
+        self.__controller.dump(req.chute)
+        return HopperDumpResponse(error=GogoRoboError(success=True))
```
Comment (Schwo0ps) [inline miso_peripherals/src/miso_peripherals/hopper_host.py:21]: I thought `success` was deprecated.

File: miso_peripherals/src/miso_peripherals/ram_hopper.py [hunk 3]
```diff
@@ -37,3 +48,17 @@ def dump(self, chute):
...
                 PULSE_DURATION,
             )
+
+
+class RAMHopperSimulator(Hopper):
+
+    def __init__(self):
+        """Create a new RAMHopperSimulator."""
+
+    def dump(self, chute):
```
Comment (Schwo0ps) [inline miso_peripherals/src/miso_peripherals/ram_hopper.py:58]: What I do for FakeLogger is `def dump(self, *args, **kwargs):`. Means this doesn't also have to be updated when the real interface changes.

Comment (rsinnet) [reply miso_peripherals/src/miso_peripherals/ram_hopper.py:58]: Oh interesting. So you do this for the children? Hmm... I could do that.

**Comment by rsinnet** [comment]
**Merge after cutting release.**

**Comment by rsinnet** [comment]
This works in sim.

File: miso_peripherals/src/miso_peripherals/hopper_host.py (hunk 2 above)
Comment (rsinnet) [reply miso_peripherals/src/miso_peripherals/hopper_host.py:21]: 🤯 I need to fix this shit. LOL.

Comment (rsinnet) [reply miso_peripherals/src/miso_peripherals/hopper_host.py]: Fixed.

File: miso_peripherals/src/miso_peripherals/ram_hopper.py (hunk 3 above)
Comment (rsinnet) [reply miso_peripherals/src/miso_peripherals/ram_hopper.py]: Fixed.

**Comment by rsinnet** [comment]
Looking over the code, shouldn't need a live test.

**Pull request merged**